test/test_other.py

```diff
@@ -15895,3 +15895,48 @@ def test_mainScriptUrlOrBlob(self, es6):
     # Now create foo.[m]js and the program should run as expected.
     shutil.copy(outfile, ('foo.%s' % ext))
     self.assertContained('hello, world', self.run_js(outfile))
+
+  @parameterized({
+    '': ([],),
+    'node': (['-sENVIRONMENT=node'],),
+  })
+  def test_locate_file_abspath(self, args):
+    # Verify that `scriptDirectory` is an absolute path
+    create_file('pre.js', '''
+      Module['locateFile'] = (fileName, scriptDirectory) => {
+        assert(nodePath['isAbsolute'](scriptDirectory), `scriptDirectory (${scriptDirectory}) should be an absolute path`);
+        return scriptDirectory + fileName;
+      };
+      ''')
+    self.do_runf('hello_world.c', 'hello, world!', emcc_args=['--pre-js', 'pre.js'] + args)
+
+  @parameterized({
+    '': ([],),
+    'node': (['-sENVIRONMENT=node'],),
+  })
+  def test_locate_file_abspath_esm(self, args):
+    # Verify that `scriptDirectory` is an absolute path when `EXPORT_ES6`
+    create_file('pre.js', '''
+      Module['locateFile'] = (fileName, scriptDirectory) => {
+        assert(nodePath['isAbsolute'](scriptDirectory), `scriptDirectory (${scriptDirectory}) should be an absolute path`);
+        return scriptDirectory + fileName;
+      };
+      ''')
+    self.run_process([EMCC, '-o', 'hello_world.mjs',
+                      '--pre-js', 'pre.js',
+                      '--extern-post-js', test_file('modularize_post_js.js'),
+                      test_file('hello_world.c')] + args)
+    self.assertContained('hello, world!', self.run_js('hello_world.mjs'))
```

Comment:
Can you use `do_runf` here to?

Reply:
It does yes.  You can enable it either by passing `-sEXPORT_ES6` explicitly in which case the underlying `build` function will choose as filename that ends in `.mjs`, or you can add `output_suffix='.mjs'`.  Either way should work.

Reply:
Great! Simplified to `do_runf()` with commit 46dd3f7c668e7886cb288360448f4383d9fc5290.